packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/data_sources/remote/interceptors/auth_interceptor.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2021, Prime Holding JSC
+// https://www.primeholding.com
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:dio/dio.dart';
+import '../../../common_use_cases/fetch_access_token_use_case.dart';
+import '../../../common_use_cases/logout_use_case.dart';
+
+/// Interceptors are a simple way to intercept and modify http requests globally
+/// before they are sent to the server. That allows us to configure
+/// authentication tokens, add logs of the requests,
+/// add custom headers and much more.
+/// Interceptors can perform a variety of implicit tasks, from authentication
+/// to logging, for every HTTP request/response. Without interception, we will
+/// have to implement these tasks explicitly for each HttpClient method call.
+/// The AuthInterceptor will inject a Token in headers.
+/// This will allow us to remain logged in on the server side or check
+/// if the token exists and allow further REST api calls.
+/// Here is an example of AuthInterceptor. You have to implement
+/// your own logic, regarding needs of your application.
+class AuthInterceptor extends Interceptor {
+  AuthInterceptor(
+    this._logoutUseCase,
+    this._fetchAccessTokenUseCase,
+  );
+
+  final LogoutUseCase _logoutUseCase;
+  final FetchAccessTokenUseCase _fetchAccessTokenUseCase;
+  final Dio _httpClient = Dio();
```

Comment:
Always inject your dependencies. In fact there should be just dio one instance. Have you checked where we don't have access to the dio instance of the interceptor itself?